include/clang/AST/Expr.h

```diff
@@ -4514,6 +4514,120 @@ class GenericSelectionExpr : public Expr {
   friend class ASTStmtReader;
 };
 
+/// \brief Represents a Checked C bounds expression.
+class BoundsExpr : public Expr {
+private:
+  SourceLocation StartLoc, RParenLoc;
```

Comment:
I tend to avoid multiple declarators like this, but I'll keep my mouth shut if this fits in with the style of this code base :)